packages/react-components/react-list-preview/library/src/components/ListItem/ListItem.types.ts

```diff
@@ -7,13 +7,17 @@ export type ListItemSlots = {
   checkmark?: Slot<typeof Checkbox>;
 };
 
+export type ListItemValue = string | number;
+
+export type ListItemOnActionData = { value: ListItemValue };
+
 /**
  * ListItem Props
  */
 export type ListItemProps = ComponentProps<ListItemSlots> & {
-  value?: string | number;
+  value?: ListItemValue;
   // eslint-disable-next-line @nx/workspace-consistent-callback-type -- using custom event here with no data
```

Comment:
Let's remove this exception and use `EventHandler` like in other components, too